agent/src/Ingest.php

```diff
@@ -23,7 +23,7 @@ class Ingest
 {
     private int $concurrentRequests = 0;
 
-    private ?TimerInterface $flushBufferAfterDelayTimer = null;
+    private ?TimerInterface $sendBufferAfterDelayTimer = null;
```

Comment:
In this PR I've added some affordances to clear buffers without having the data prepared in anyway.

I realised that `flush` was kinda used incorrectly throughout the codebase. I was using `flush` as a way to gather the stream data and clear it.

Now we use `flush` to indicate that the data in the stream is thrown away.

We use `read` to indicate that the data is pulled off the stream.

So there is a bit of renaming throughout the PR to make sure we use the same language everywhere.